### DISCUSSION TITLE
Problem inserting "book variables" into rendered page

### DISCUSSION THREAD

**Discussion by hmaier-fws** [honkit/honkit#340] [Q&A]
How can the **book variables** be inserted into a rendered page? The [variables documentation](https://honkit.netlify.app/templating/variables.html) states that is a "...reference of the *available data during book's parsing*...". I am attempting to insert several [Book Variables](https://honkit.netlify.app/templating/variables.html#book-variables), but they are not being displayed in the rendered page.

For example, my book.json contains a `title` key with an associated value that is being read and rendered for use by the html title tag. When I attempt to add `book.title` within the body of a page it does not render to the final html. I have tried all of the following in my source .md but none render:

  - {{ title }}
  - {{ book.title }}
  - {{ gitbook.title }}
  - {{ this.title }}
 
Note: I tried `this.title` because when serving the page locally I receive a warning: *"book" property is deprecated, use "this" directly instead*
 
I am using honkit v. 4.0.7, running under NodeJS v. 18.15.0 on Windows 10.

Any suggestions would be appreciated!

**Comment by azu** [comment]
Can you paste your `book.json` file? 

`{{ book.title }}` will refer to `book.json`'s `variables.title`.

```json
{
  "variables": {
    "title": "xxx"
  }
}
```

**Reply by hmaier-fws** [reply]
My "title" is not a property of the the variables object (`variables.title`), but a property of the book object (`book.title`). Here are the contents of my book.json:

> {
> title: "My title from book.json",
> pubdate: "2023-04-24",
> root: "./"
> }

My use seems to be consistent with how the other [*general settings*](https://honkit.netlify.app/config.html#general-settings) properties, described in the [HonKit Toolchain Documentation](https://honkit.netlify.app/), are used in several book.json files referenced by the HonKit project, such as:
  - `book.root` as used in the [multiple-languages/book.json](https://github.com/honkit/honkit/blob/master/examples/multiple-languages/book.json) file
  - `book.title`, and several other general settings, as used in the [benchmark/book.json](https://github.com/honkit/honkit/blob/master/examples/benchmark/book.json) and the [misp-book/book.json](https://github.com/MISP/misp-book/blob/main/book.json) files

As I mentioned above, the `book.title` **is being properly read** by HonKit when rendering the book because **it is used** to set the html page title tag. It **is not** being used when I attempt to use it as a variable and insert it into a page as described by the [HonKit Templating guidance](https://honkit.netlify.app/templating/). The documentation states that "*HonKit provides a set of [predefined variables](https://honkit.netlify.app/templating/variables.html) from the context*", which implies that the referenced ("predefined") variables should be available for use at the page-level. Is that not the case?
